File: src/compiler/typescript_to_rust/generators/ic_object/functions/method_name.ts
import { Rust } from '../../../../../types';

export function generateIcObjectFunctionMethodName(): Rust {
    return /* rust */ `
        fn _azle_ic_method_name(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            _context: &mut boa_engine::Context
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            Ok(ic_cdk::api::call::method_name().into())
        }
    `;
}

File: src/compiler/typescript_to_rust/azle_generate_rearchitecture/src/ic/method_name.rs
use quickjs_wasm_rs::{to_qjs_value, CallbackArg, JSContextRef, JSValue, JSValueRef};

pub fn native_function<'a>(
    context: &'a JSContextRef,
    _this: &CallbackArg,
    _args: &[CallbackArg],
) -> Result<JSValueRef<'a>, anyhow::Error> {
    let method_name_js_value: JSValue = ic_cdk::api::call::method_name().into();
    to_qjs_value(&context, &method_name_js_value)
}
